docs/userguide/storagedriver/btrfs-driver.md

```diff
@@ -0,0 +1,266 @@
+
+        $ sudo apt-get install btrfs-tools
+        Reading package lists... Done
+        Building dependency tree
+        <output truncated>
+
+2. Create the Btrfs storage pool.
+
+    Btrfs storage pools are created with the `mkfs.btrfs` command. Passing multiple devices to the `mkfs.btrfs` command creates a pool across all of those devices. Here you create a pool with single device at `/dev/xvdb`.
+
+        $ sudo mkfs.btrfs -f /dev/xvdb
+        WARNING! - Btrfs v3.12 IS EXPERIMENTAL
+        WARNING! - see http://btrfs.wiki.kernel.org before using
+
+        Turning ON incompat feature 'extref': increased hardlink limit per file to 65536
+        fs created label (null) on /dev/xvdb
+            nodesize 16384 leafsize 16384 sectorsize 4096 size 4.00GiB
+        Btrfs v3.12
+
+    Be sure to substitute `/dev/xvdb` with the appropriate device(s) on your
+    system.
+
+    > **Warning**: Take note of the warning about Btrfs being experimental. As
+    noted earlier, Btrfs is not currently recommended for production deployments
+    unless you already have extensive experience.
+
+3. If it does not already exist, create a directory for the Docker host's local storage area at `/var/lib/docker`.
+
+        $ sudo mkdir /var/lib/docker
+
+4. Configure the system to automatically mount the Btrfs filesystem each time the system boots.
+
+    a. Obtain the Btrfs filesystem's UUID.
+
+        $ sudo blkid /dev/xvdb
+        /dev/xvdb: UUID="a0ed851e-158b-4120-8416-c9b072c8cf47" UUID_SUB="c3927a64-4454-4eef-95c2-a7d44ac0cf27" TYPE="btrfs"
+
+    b. Create a `/etc/fstab` entry to automatically mount `/var/lib/docker` each time the system boots.
+
+        /dev/xvdb /var/lib/docker btrfs defaults 0 0
+        UUID="a0ed851e-158b-4120-8416-c9b072c8cf47" /var/lib/docker btrfs defaults 0 0
+
+5. Mount the new filesystem and verify the operation.
+
+        $ sudo mount -a
+        $ mount
+        /dev/xvda1 on / type ext4 (rw,discard)
+        <output truncated>
+        /dev/xvdb on /var/lib/docker type btrfs (rw)
+
+    The last line in the output above shows the `/dev/xvdb` mounted at `/var/lib/docker` as Btrfs.
+
+
+Now that you have a Btrfs filesystem mounted at `/var/lib/docker`, the daemon should automatically load with the `btrfs` storage driver.
+
+1. Start the Docker daemon.
+
+        $ sudo service docker start
+        docker start/running, process 2315
+
+    The procedure for starting the Docker daemon may differ depending on the
+    Linux distribution you are using.
+
+    You can start the Docker daemon ith the `btrfs` storage driver by passing
+    the `--storage-driver=btrfs` flag to the `docker daemon` command or you can
+    add the `DOCKER_OPTS` line to the Docker config file.
+
+2. Verify the storage driver with the `docker info` command.
+
+        $ sudo docker info
+        Containers: 0
+        Images: 0
+        Storage Driver: btrfs
+        [...]
+
+Your Docker host is now configured to use the `btrfs` storage driver.
```

Comment:
Just thinking of this, because we've had some issues with switching between storage drivers; It may be important to mention that people should delete all their containers before switching, otherwise containers of the previous storage driver could be still available in the "graph" database, but not able to be started because the containers themselves are using the previous storage driver